### PR TITLE
Support adding and editing the base configuration of an observation

### PR DESCRIPTION
**Pull request by cquiroz**
For details see `docs/ScienceConfiguration.md`
We way want to add an `editScienceConfiguration` operation as well

### CODE REVIEW

File: modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala [hunk 1]
```diff
@@ -34,7 +36,8 @@ final case class ObservationModel(
   constraintSet:        ConstraintSetModel,
   scienceRequirements:  ScienceRequirements,
   plannedTimeSummary:   PlannedTimeSummaryModel,
-  config:               Option[InstrumentConfigModel.Reference]
+  config:               Option[InstrumentConfigModel.Reference],
```
Comment (swalker2m) [inline modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala:39]: Maybe let's keep `config` last?  At some point I may want to figure out a new name.  Maybe `manualConfig`.

File: modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala [hunk 2]
```diff
@@ -140,44 +146,56 @@ object ObservationModel extends ObservationOptics {
...
+    val pointing: ValidatedInput[(Input[Asterism.Id], Input[Target.Id])] =
       (asterismId, targetId) match {
         case (Assign(_), Assign(_)) => InputError.fromMessage(s"Cannot assign both an asterism and a target to the observation").invalidNec
         case _                      => (asterismId, targetId).validNec
       }
 
+    def editOrCreateSciConfig(ed: Option[Option[Either[ScienceConfigurationModel, State[ScienceConfigurationModel, Unit]]]])(m: Option[ScienceConfigurationModel]): Option[ScienceConfigurationModel] =
+      ed.flatMap {
+        case Some(Left(s))  => s.some
+        case Some(Right(s)) => m.map(s.runS(_).value)
```
Comment (swalker2m) [inline modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala:173]: If you choose to edit but the science configuration doesn't exist yet, then essentially nothing happens correct?

**Review by swalker2m — APPROVED** [review]
Well it all looks perfectly cromulent to me.  Good work figuring out this mess! :+1:

File: modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala (hunk 1 above)
Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala:39]: But no need to change it now, I can move it later.

File: modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala (hunk 2 above)
Comment (cquiroz) [reply modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala:173]: Right, It is ignored, as we have discussed an alternative would be for it to fail in that case

Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala:173]: Let's go with this as I'm not sure what the "right" answer is anyway.

**Pull request merged**